### PR TITLE
fix: validate rules in Apisix Dashboard is same as Apisix api

### LINKED ISSUE #2057: Make sure validate rules in Apisix Dashboard is same as Apisix api
## Issue description

Some rules can be created by CRD, but can not be modified in dashboard.

e.g. `/` in ApisixRoute.spec.http[].name

image

e.g. add key in limit-count.lua

image

## Expected behavior

Make sure validate rules in Apisix Dashboard is same as Apisix api.

### PR DESCRIPTION
**Pull request by iamayushdas**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [x] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with detailed description.

**Related issues**

fix/resolve #2057 

**Checklist:**

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by iamayushdas** [comment]
cc @liuxiran @guoqqqi @juzhiyuan

File: docs/en/latest/I18N_USER_GUIDE.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ we have already defined many global keys, before you do i18n, you can refer to [
 
 ```js
 'page.route.form.itemRulesExtraMessage.parameterName': '仅支持字母和数字，且只能以字母开头',
-'page.route.form.itemRulesPatternMessage.apiNameRule': '最大长度100，仅支持字母、数字、- 和 _，且只能以字母开头',
+'page.route.form.itemRulesPatternMessage.apiNameRule': '最大长度应仅为 100',
```
Comment (foolwc) [inline docs/en/latest/I18N_USER_GUIDE.md:67]:
``` suggestion
'page.route.form.itemRulesPatternMessage.apiNameRule': '最大长度应小于100',
```

**Comment by liuxiran** [comment]
marked an unstable test case here: https://github.com/apache/apisix-dashboard/pull/2085/checks?check_run_id=3410472747

**Comment by liuxiran** [comment]
Hi @iamayushdas , would you please add test case for your changes? 

just add assertion to check a route name with 101 characters can be added to this test case, https://github.com/apache/apisix-dashboard/blob/fb789638b30019e10f9d0359bebee01de8bae125/web/cypress/integration/route/create-edit-duplicate-delete-route.spec.js#L86

**Comment by iamayushdas** [comment]
Sure sure

**Review by liuxiran — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**